tests/integrations/gateway_fixtures.py

```diff
@@ -0,0 +1,93 @@
+import typing as t
+
+import boto3
+import pytest
+import requests
+import scaleway.container.v1beta1 as sdk
+from scaleway import Client
+
+from tests import constants
+
+from .utils import create_client
+
+
+@pytest.fixture()
+def api_gateway(scaleway_project: str) -> t.Iterator[sdk.Container]:
+    client = create_client()
+    container = None
+    try:
+        container = _deploy_gateway(scaleway_project, client)
+        yield container
+    finally:
+        if container:
+            _cleanup_gateway(client, container)
+
+
+@pytest.fixture()
+def auth_key(
+    api_gateway: sdk.Container,  # pylint: disable=redefined-outer-name
+) -> str:
+    client = create_client()
+
+    response = requests.post(
+        "https://" + api_gateway.domain_name + "/token",
+        timeout=constants.COLD_START_TIMEOUT,
+    )
+    response.raise_for_status()
+
+    s3 = boto3.resource(
+        "s3",
+        region_name=constants.API_GATEWAY_S3_BUCKET,
+        endpoint_url=constants.API_GATEWAY_S3_BUCKET_ENDPOINT,
+        aws_access_key_id=client.access_key,
+        aws_secret_access_key=client.secret_key,
+    )
+
+    objects = sorted(
+        s3.Bucket(constants.API_GATEWAY_S3_BUCKET).objects.all(),  # type: ignore
+        key=lambda obj: obj.last_modified,
+        reverse=True,
+    )
+    key = objects[0].key
+    return key
+
+
+def _deploy_gateway(project_id: str, client: Client) -> sdk.Container:
```

Comment:
I am worried about duplicating the gateway logic in this test. Instead, it might be cleaner and safer to have a setup step in the integration tests which did the following:

1. Check out a tag of the API gateway
2. Run the relevant `make` commands to set up the gateway
3. Run a (new?) `make` command to get any valid auth key
4. Run another (new?) `make` command to get the gateway URL
5. Export this auth key and gateway URLs as environment variables to be used in this test

As I've noted, I'm not sure if the `make` interface in the gateway supports this, but it would be nice if it did. If it works, this could then become a reusable Github Action to set up and use your own gateway.